demos/mqtt/mqtt_demo_plaintext/mqtt_demo_plaintext.c

```diff
@@ -423,6 +438,43 @@ static int connectToServer( const char * pServer,
 
 /*-----------------------------------------------------------*/
 
+static int connectToServerWithBackoffRetries( int * pTcpSocket )
+{
+    int status = EXIT_SUCCESS;
+    bool backoffSuccess = true;
+
+    /* Initialize reconnect attempts and interval */
+    reconnectBackoffReset();
+
+    /* Attempt to connect to MQTT broker. If connection fails, retry after
+     * a timeout. Timeout value will exponentially increase till maximum
+     * attemps are reached.
+     *
+     *  while( ( EXIT_FAILURE == connectToServer( BROKER_ENDPOINT, BROKER_PORT, pTcpSocket ) ) &&
+     *  ( true == reconnectBackoffAndSleep() ) ) {}
+     */
+
+    do
+    {
+        status = connectToServer( BROKER_ENDPOINT, BROKER_PORT, pTcpSocket );
```

Comment:
We can possibly have multiple `connectToServer` functions, so we can probably have an interface for this.

Reply:
In past we discussed to have an interface using function pointers, but during the design review, it was discouraged. connectToServerWithBackoffRetries() will be implemented by the application using reconnectBackoffReset() and reconnectBackoffAndSleep()